[ML] APM Latency Correlations UX improvements/fixes

walterra opened the pull request:
## Summary

Part of #108332.
Follow up to #107266.

- Tweak label for log log chart data series for all transactions. f53d428dbe0
- Show tip to drag to select a range in the Trace samples distribution chart panel. 086c5a0df5f
- Change the selected range label and clear selection option to `EuiBadge` with click action and tooltip on hover 'Clear selection'. 64937ff0578
- Move the 'current sample' annotation to the bottom axis to not overlap with the 95p annotation marker. f50e96f1c64
- Move the help popover to the top of the panel. f24a5cca50f
- Move the refresh/cancel option to the right of the progress bar. f4253397dae
- Always display the table for correlations. 212b77f00e0
- Show improved empty state using the `EuiEmptyPrompt` for the correlations table when no significant correlations are found. 212b77f00e0
- Fix Filter behaviour: when clicking on the filter item, jump to show top of page where the filter bar is evident, and reset the correlation tabs and show the Trace Samples tab ebc05ae84a8
- Fix updating the log log chart and correlation service when the user changes the kquery or time span. 5804216

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)



### For maintainers

- [x] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

walterra committed:
[ML] Draft version of correlations being added to main pages.

walterra committed:
[ML] Add selection capability to latency histogram chart.

walterra committed:
[ML] Adds button to toggle correlations visibility.

walterra committed:
[ML] Improve chart refresh, adds option to load overall histogram only.

walterra committed:
remove percentileThresholdValue from client side params.

walterra committed:
[ML] More fine grained latency range selection.

walterra committed:
[ML] Fix jest tests.

walterra committed:
[ML] Clear selection button.

walterra committed:
[ML] Fix to clear selection.

walterra committed:
[ML] Return trace samples based on chart selection.

walterra committed:
Merge branch 'master' into ml-apm-correlations-page

walterra committed:
[ML] Move transaction content to tabs.

walterra committed:
[ML] Fix correlation chart loading state.

walterra committed:
[ML] Fix translation ids.

walterra committed:
[ML] Selection UI tweaks.

walterra committed:
[ML] Weight retainer tweak.

walterra committed:
[ML] Adds current sample as marker to correlations chart.

walterra committed:
[ML] Tweak y axis ticks.

walterra committed:
[ML] Tweak EuiPanel layout. Fix translations ids.

walterra committed:
[ML] Set default table rows to 10.

walterra committed:
[ML] Fix i18n.

walterra committed:
[Ml] Remove outdated code.

walterra committed:
[ML] Disable drag select for correlation analysis.

walterra committed:
[ML] Tweak vertical alignment.

walterra committed:
[ML] Refactor to fetch samples without buckets only.

walterra committed:
[ML] Fix i18n.

walterra committed:
Merge branch 'master' into ml-apm-correlations-page

walterra committed:
Merge branch 'master' into ml-apm-correlations-page

walterra committed:
[ML] Fix async search service to use state provider.

walterra committed:
[ML] Tweak trace samples API response. Fix API integration tests.

walterra committed:
[ML] Refactor tabs.

walterra committed:
[ML] Add license gate to failed transactions.

walterra committed:
[ML] Remove 'Correlations' button.

walterra committed:
Merge branch 'master' into ml-apm-correlations-page

walterra committed:
[ML] Readd beta badge.

walterra committed:
[ML] Fix i18n.

walterra committed:
[ML] Remove flyout based correlations component.

walterra committed:
[ML] Rename correlations tab.

walterra committed:
[ML] Move beta badge to tab.

walterra committed:
[ML] Fix i18n.

walterra committed:
[ML] Update functional tests.

walterra committed:
Revert "[ML] Tweak y axis ticks."

This reverts commit cc2479e385411ae3b17cb84d28031767c37d6021.

walterra committed:
Merge branch 'master' into ml-apm-correlations-page

walterra committed:
[ML] Tweak annotation id.

walterra committed:
[ML] Refactor to get rid of 'params'.

walterra committed:
[ML] Rename failed transactions tab.

walterra committed:
[ML] Refactor to have separate components for trace samples and latency correlations.

walterra committed:
[ML] Tweak url parameters.

walterra committed:
[ML] Fix log-log chart title.

walterra committed:
[ML] Fetch up to 500 trace samples instead of just 10.

walterra committed:
Merge branch 'master' into ml-apm-correlations-page

walterra committed:
Merge branch 'master' into ml-apm-correlations-page

walterra committed:
[ML] Fix tests.

walterra committed:
Merge branch 'master' into ml-apm-correlations-page

walterra committed:
[ML] Fix assertion.

walterra committed:
[ML] Remove Array.from().

walterra committed:
[ML] Tweak var names related to percentiles.

walterra committed:
Merge branch 'master' into ml-apm-correlations-page

walterra committed:
[ML] Simplify code to find selectedSample

walterra committed:
[ML] Fix waterfall message if no trace samples can be found. Fix clear selection to display correct annotation and first sample.

walterra committed:
Merge branch 'ml-apm-correlations-page' into ml-apm-correlations-page-ux

walterra committed:
[Ml] Tweak label for log log chart data series for all transactions.

walterra committed:
[ML] Show tip to drag to select a range in the Trace samples distribution chart panel.

walterra committed:
[ML] Change the selected range label and clear selection option to EuiBadge with click action and tooltip on hover 'Clear selection'.

walterra committed:
[ML] Move the 'current sample' annotation to the bottom axis to not overlap with the 95p annotation marker.

walterra committed:
[ML] Move the help popover to the top of the panel.

walterra committed:
[ML] Move the refresh/cancel option to the right of the progress bar.

walterra committed:
[ML] Correlations table tweaks.

formgeist commented:
@walterra Looks great, thanks for including these.

**Badge displaying selected area in distribution**

![CleanShot 2021-08-17 at 13 04 04](https://user-images.githubusercontent.com/4104278/129715407-3ee0c25c-b444-45ea-9237-2dc7a6429a12.gif)

Would be good to change the `onClick` to also allow for clicking the text inside to remove the selection - it will help the click-area of the badge. Described how to change the onClick event here https://elastic.github.io/eui/#/display/badge#badge-with-onclick-events

**Correlations table empty state copy**

![CleanShot 2021-08-17 at 13 07 58@2x](https://user-images.githubusercontent.com/4104278/129715900-e3a50ce4-9f04-426a-95b4-3a333dc54435.png)

Small nit; but perhaps we can word-break to the next line for the "Try..." copy

Otherwise, it looks great 🎉

walterra committed:
[ML] Tweak filter behavior.

walterra committed:
[ML] Fix clear selection badge click.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146091/)
* Commit: 212b77f00e05ad8816345a1256dd7d777e9042ee
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-108860/212b77f00e05ad8816345a1256dd7d777e9042ee)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup4](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146091/execution/node/675/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146091/execution/node/714/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146091/","number":"146091","commit":"212b77f00e05ad8816345a1256dd7d777e9042ee"}]}
PIPELINE-->

walterra committed:
[ML] Enforce line break.

walterra commented:
- Fixed the `onClick` area of the clear selection badge in 30c9ee83bb9
- Enforced a line break for the no significant correlations found text in 8ca296c

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146116/)
* Commit: 8ca296c2dfa8741a9630ae78120dac122fd1041a
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146116/execution/node/421/log)

### History
* :broken_heart: [Build #146091](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146091/) failed 212b77f00e05ad8816345a1256dd7d777e9042ee

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146091/","number":"146091","commit":"212b77f00e05ad8816345a1256dd7d777e9042ee"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146116/","number":"146116","commit":"8ca296c2dfa8741a9630ae78120dac122fd1041a"}]}
PIPELINE-->

walterra committed:
[ML] Fix i18n.

formgeist commented:
@walterra One more thing and this a general observation for when changing the query either by updating the time range selected or adding a filter to the query bar (e.g. a correlation field and value), the metrics are updated but the distribution stays the same and the correlation job is not run again. I assume we'd want to at least update the overall distribution and re-run the correlations analysis on the new selected data set.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146107/)
* Commit: ebc05ae84a82a1a4d4f180ae4acb601bf808c48d
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-108860/ebc05ae84a82a1a4d4f180ae4acb601bf808c48d)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146107/execution/node/821/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146107/execution/node/893/log)

### History
* :broken_heart: [Build #146116](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146116/) failed 8ca296c2dfa8741a9630ae78120dac122fd1041a
* :broken_heart: [Build #146091](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146091/) failed 212b77f00e05ad8816345a1256dd7d777e9042ee

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146091/","number":"146091","commit":"212b77f00e05ad8816345a1256dd7d777e9042ee"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146116/","number":"146116","commit":"8ca296c2dfa8741a9630ae78120dac122fd1041a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146107/","number":"146107","commit":"ebc05ae84a82a1a4d4f180ae4acb601bf808c48d"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146129/)
* Commit: 69a4874074e59eed659360c0e77fbbb58576a32e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-108860/69a4874074e59eed659360c0e77fbbb58576a32e)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Integration Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146129/execution/node/441/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146129/execution/node/714/log)

### History
* :broken_heart: [Build #146116](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146116/) failed 8ca296c2dfa8741a9630ae78120dac122fd1041a
* :broken_heart: [Build #146107](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146107/) failed ebc05ae84a82a1a4d4f180ae4acb601bf808c48d
* :broken_heart: [Build #146091](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146091/) failed 212b77f00e05ad8816345a1256dd7d777e9042ee

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146091/","number":"146091","commit":"212b77f00e05ad8816345a1256dd7d777e9042ee"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146107/","number":"146107","commit":"ebc05ae84a82a1a4d4f180ae4acb601bf808c48d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146116/","number":"146116","commit":"8ca296c2dfa8741a9630ae78120dac122fd1041a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146129/","number":"146129","commit":"69a4874074e59eed659360c0e77fbbb58576a32e"}]}
PIPELINE-->

walterra committed:
Merge branch 'master' into ml-apm-correlations-page-ux

walterra committed:
Merge branch 'master' into ml-apm-correlations-page-ux

walterra committed:
Merge branch 'master' into ml-apm-correlations-page-ux

walterra committed:
[ML] Fix log log chart and correlations service reloading.

walterra commented:
@formgeist fixed the behavior to update the log log chart and restart correlations analysis when the user changes the time span in 5804216.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146239/)
* Commit: 84b05bcee6ddedb8ad7208aecf76fdfda764f14d
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-108860/84b05bcee6ddedb8ad7208aecf76fdfda764f14d)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146239/execution/node/651/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146239/execution/node/839/log)

### History
* :broken_heart: [Build #146129](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146129/) failed 69a4874074e59eed659360c0e77fbbb58576a32e
* :broken_heart: [Build #146116](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146116/) failed 8ca296c2dfa8741a9630ae78120dac122fd1041a
* :broken_heart: [Build #146107](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146107/) failed ebc05ae84a82a1a4d4f180ae4acb601bf808c48d
* :broken_heart: [Build #146091](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146091/) failed 212b77f00e05ad8816345a1256dd7d777e9042ee

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146091/","number":"146091","commit":"212b77f00e05ad8816345a1256dd7d777e9042ee"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146107/","number":"146107","commit":"ebc05ae84a82a1a4d4f180ae4acb601bf808c48d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146116/","number":"146116","commit":"8ca296c2dfa8741a9630ae78120dac122fd1041a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146129/","number":"146129","commit":"69a4874074e59eed659360c0e77fbbb58576a32e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146239/","number":"146239","commit":"84b05bcee6ddedb8ad7208aecf76fdfda764f14d"}]}
PIPELINE-->

walterra committed:
[ML] Update functional tests.

qn895 commented on x-pack/plugins/apm/public/components/app/transaction_details/transaction_details_tabs.tsx:
```diff
@@ -81,6 +81,15 @@ export function TransactionDetailsTabs() {
 
+  // When filtering in either the latency correlations or failed transactions correlations tab,
+  // scroll to the top of the page and switch to the 'Trace samples' tab to trigger a refresh.
+  const onFilter = () => {
```
nit: this can be called in a callback hook

walterra replied:
Fixed in d5bf6b55181.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146261/)
* Commit: 5804216909f9decb3b4521d01ea5095b7a7f2322
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-108860/5804216909f9decb3b4521d01ea5095b7a7f2322)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146261/execution/node/671/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146261/execution/node/840/log)

### History
* :broken_heart: [Build #146239](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146239/) failed 84b05bcee6ddedb8ad7208aecf76fdfda764f14d
* :broken_heart: [Build #146129](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146129/) failed 69a4874074e59eed659360c0e77fbbb58576a32e
* :broken_heart: [Build #146116](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146116/) failed 8ca296c2dfa8741a9630ae78120dac122fd1041a
* :broken_heart: [Build #146107](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146107/) failed ebc05ae84a82a1a4d4f180ae4acb601bf808c48d
* :broken_heart: [Build #146091](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146091/) failed 212b77f00e05ad8816345a1256dd7d777e9042ee

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146091/","number":"146091","commit":"212b77f00e05ad8816345a1256dd7d777e9042ee"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146107/","number":"146107","commit":"ebc05ae84a82a1a4d4f180ae4acb601bf808c48d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146116/","number":"146116","commit":"8ca296c2dfa8741a9630ae78120dac122fd1041a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146129/","number":"146129","commit":"69a4874074e59eed659360c0e77fbbb58576a32e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146239/","number":"146239","commit":"84b05bcee6ddedb8ad7208aecf76fdfda764f14d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146261/","number":"146261","commit":"5804216909f9decb3b4521d01ea5095b7a7f2322"}]}
PIPELINE-->

qn895 commented on x-pack/plugins/apm/public/components/app/correlations/latency_correlations.tsx:
```diff
@@ -86,29 +86,38 @@ export function LatencyCorrelations({ onFilter }: { onFilter: () => void }) {
+      percentileThreshold: DEFAULT_PERCENTILE_THRESHOLD,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [kuery, start, end]);
```
This should also depends on other parts of the params like `serviceName` or `environmentName`

walterra replied:
Fixed in 0762e556874

walterra committed:
[Ml] Fix attributes for changes check.

formgeist commented:
@walterra I noticed two more things just going through it again (in dark mode).

**Default charts configuration is not used**
APM uses a default configuration and theme provider (through EUI) for supporting dark theme and I believe configurations around the e.g. gridlines in the chart canvas. I could see in the distribution chart the appropriate theming isn't added for dark theme and the gridlines are not present (hard to tell on the light theme).

<img width="1449" alt="CleanShot 2021-08-17 at 21 12 46@2x" src="https://user-images.githubusercontent.com/4104278/129786473-672089ce-6ffe-4aa3-96cd-1e3b1291a747.png">

**Filters added using special chars are not formatted correctly in the query bar**
I added a filter for a `url.path` and it didn't return any results in the data because the formatting from clicking the filter value option in the table wasn't correct.

<img width="1455" alt="CleanShot 2021-08-17 at 20 29 58@2x" src="https://user-images.githubusercontent.com/4104278/129786601-c1c531db-ba19-4d86-a71d-48af318af99b.png">

<img width="1452" alt="CleanShot 2021-08-17 at 20 30 15@2x" src="https://user-images.githubusercontent.com/4104278/129786647-916fcd30-1007-4ba1-982a-282f15a5fc7e.png">

<img width="1461" alt="CleanShot 2021-08-17 at 20 30 30@2x" src="https://user-images.githubusercontent.com/4104278/129786685-f5c113cb-f743-4d37-9b98-104211d4e75e.png">

_Fixed the formatting of the URL yielded results in the end_

walterra committed:
[ML] Fix onFilter.

kibanamachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146272/)
* Commit: d908b7ec7968e87f111a2b75b2955e9dda551929
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-108860/d908b7ec7968e87f111a2b75b2955e9dda551929)

* Flaky suites:
  * xpack-APMCypress

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/d07f7a5d5e30e8ca272413761cdfe3be1508a34b) | [after](https://github.com/elastic/kibana/commit/d908b7ec7968e87f111a2b75b2955e9dda551929) | diff |
| --- | --- | --- | --- |
| `apm` | 4.4MB | 4.4MB | +2.2KB |

### History
* :broken_heart: [Build #146261](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146261/) failed 5804216909f9decb3b4521d01ea5095b7a7f2322
* :broken_heart: [Build #146239](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146239/) failed 84b05bcee6ddedb8ad7208aecf76fdfda764f14d
* :broken_heart: [Build #146129](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146129/) failed 69a4874074e59eed659360c0e77fbbb58576a32e
* :broken_heart: [Build #146116](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146116/) failed 8ca296c2dfa8741a9630ae78120dac122fd1041a
* :broken_heart: [Build #146107](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146107/) failed ebc05ae84a82a1a4d4f180ae4acb601bf808c48d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146107/","number":"146107","commit":"ebc05ae84a82a1a4d4f180ae4acb601bf808c48d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146116/","number":"146116","commit":"8ca296c2dfa8741a9630ae78120dac122fd1041a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146129/","number":"146129","commit":"69a4874074e59eed659360c0e77fbbb58576a32e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146239/","number":"146239","commit":"84b05bcee6ddedb8ad7208aecf76fdfda764f14d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146261/","number":"146261","commit":"5804216909f9decb3b4521d01ea5095b7a7f2322"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146272/","number":"146272","commit":"d908b7ec7968e87f111a2b75b2955e9dda551929"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

walterra committed:
[ML] Fix chart theming.

walterra commented:
@formgeist Thanks for the hint with the theming in other charts, I was able to fix it in 14c68963aaa

Fixed Light mode:

![image](https://user-images.githubusercontent.com/230104/129792931-6b20c1a8-f56f-4806-9bfc-f69526080c4b.png)

Fixed Dark mode:

![image](https://user-images.githubusercontent.com/230104/129792725-3c9f236c-6107-42bb-ba6e-8ddf21e18bfd.png)

qn895 commented:
Latest changes LGTM 🎉

walterra committed:
Merge branch 'master' into ml-apm-correlations-page-ux

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146379/)
* Commit: 3556cfb6adf3847aa3c6b028f031f94fc5ca673c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-108860/3556cfb6adf3847aa3c6b028f031f94fc5ca673c)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-APMCypress](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146379/execution/node/577/log)
* [Execute xpack-APMCypress](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146379/execution/node/709/log)

### History
* :yellow_heart: [Build #146272](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146272/) was flaky d908b7ec7968e87f111a2b75b2955e9dda551929
* :broken_heart: [Build #146261](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146261/) failed 5804216909f9decb3b4521d01ea5095b7a7f2322
* :broken_heart: [Build #146239](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146239/) failed 84b05bcee6ddedb8ad7208aecf76fdfda764f14d
* :broken_heart: [Build #146129](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146129/) failed 69a4874074e59eed659360c0e77fbbb58576a32e
* :broken_heart: [Build #146116](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146116/) failed 8ca296c2dfa8741a9630ae78120dac122fd1041a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146116/","number":"146116","commit":"8ca296c2dfa8741a9630ae78120dac122fd1041a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146129/","number":"146129","commit":"69a4874074e59eed659360c0e77fbbb58576a32e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146239/","number":"146239","commit":"84b05bcee6ddedb8ad7208aecf76fdfda764f14d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146261/","number":"146261","commit":"5804216909f9decb3b4521d01ea5095b7a7f2322"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146272/","number":"146272","commit":"d908b7ec7968e87f111a2b75b2955e9dda551929"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146379/","number":"146379","commit":"3556cfb6adf3847aa3c6b028f031f94fc5ca673c"}]}
PIPELINE-->

walterra commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'master' into ml-apm-correlations-page-ux

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146402/)
* Commit: c652ac83dc16fc62883120bc18a46e3f6195c442
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-108860/c652ac83dc16fc62883120bc18a46e3f6195c442)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-APMCypress](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146402/execution/node/531/log)
* [Execute xpack-APMCypress](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146402/execution/node/782/log)

### History
* :broken_heart: [Build #146379](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146379/) failed 3556cfb6adf3847aa3c6b028f031f94fc5ca673c
* :yellow_heart: [Build #146272](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146272/) was flaky d908b7ec7968e87f111a2b75b2955e9dda551929
* :broken_heart: [Build #146261](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146261/) failed 5804216909f9decb3b4521d01ea5095b7a7f2322
* :broken_heart: [Build #146239](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146239/) failed 84b05bcee6ddedb8ad7208aecf76fdfda764f14d
* :broken_heart: [Build #146129](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146129/) failed 69a4874074e59eed659360c0e77fbbb58576a32e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146129/","number":"146129","commit":"69a4874074e59eed659360c0e77fbbb58576a32e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146239/","number":"146239","commit":"84b05bcee6ddedb8ad7208aecf76fdfda764f14d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146261/","number":"146261","commit":"5804216909f9decb3b4521d01ea5095b7a7f2322"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146272/","number":"146272","commit":"d908b7ec7968e87f111a2b75b2955e9dda551929"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146379/","number":"146379","commit":"3556cfb6adf3847aa3c6b028f031f94fc5ca673c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146402/","number":"146402","commit":"c652ac83dc16fc62883120bc18a46e3f6195c442"}]}
PIPELINE-->

walterra commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'master' into ml-apm-correlations-page-ux

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146421/)
* Commit: 0536f2982733706c3b7e4c73a6418311ad227aa2
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-108860/0536f2982733706c3b7e4c73a6418311ad227aa2)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-APMCypress](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146421/execution/node/531/log)
* [Execute xpack-APMCypress](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146421/execution/node/782/log)

### History
* :broken_heart: [Build #146379](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146379/) failed 3556cfb6adf3847aa3c6b028f031f94fc5ca673c
* :broken_heart: [Build #146402](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146402/) failed c652ac83dc16fc62883120bc18a46e3f6195c442
* :yellow_heart: [Build #146272](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146272/) was flaky d908b7ec7968e87f111a2b75b2955e9dda551929
* :broken_heart: [Build #146261](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146261/) failed 5804216909f9decb3b4521d01ea5095b7a7f2322
* :broken_heart: [Build #146239](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146239/) failed 84b05bcee6ddedb8ad7208aecf76fdfda764f14d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146239/","number":"146239","commit":"84b05bcee6ddedb8ad7208aecf76fdfda764f14d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146261/","number":"146261","commit":"5804216909f9decb3b4521d01ea5095b7a7f2322"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146272/","number":"146272","commit":"d908b7ec7968e87f111a2b75b2955e9dda551929"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146402/","number":"146402","commit":"c652ac83dc16fc62883120bc18a46e3f6195c442"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146379/","number":"146379","commit":"3556cfb6adf3847aa3c6b028f031f94fc5ca673c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146421/","number":"146421","commit":"0536f2982733706c3b7e4c73a6418311ad227aa2"}]}
PIPELINE-->

qn895 commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'master' into ml-apm-correlations-page-ux

sorenlouv reviewed (approved):
lgtm

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146439/)
* Commit: fd453a24cbaaaac390248960a656eeead0fe6eda
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-108860/fd453a24cbaaaac390248960a656eeead0fe6eda)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-APMCypress](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146439/execution/node/534/log)
* [Execute xpack-APMCypress](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146439/execution/node/792/log)

### History
* :broken_heart: [Build #146402](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146402/) failed c652ac83dc16fc62883120bc18a46e3f6195c442
* :broken_heart: [Build #146421](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146421/) failed 0536f2982733706c3b7e4c73a6418311ad227aa2
* :broken_heart: [Build #146379](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146379/) failed 3556cfb6adf3847aa3c6b028f031f94fc5ca673c
* :yellow_heart: [Build #146272](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146272/) was flaky d908b7ec7968e87f111a2b75b2955e9dda551929
* :broken_heart: [Build #146261](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146261/) failed 5804216909f9decb3b4521d01ea5095b7a7f2322

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146261/","number":"146261","commit":"5804216909f9decb3b4521d01ea5095b7a7f2322"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146272/","number":"146272","commit":"d908b7ec7968e87f111a2b75b2955e9dda551929"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146379/","number":"146379","commit":"3556cfb6adf3847aa3c6b028f031f94fc5ca673c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146421/","number":"146421","commit":"0536f2982733706c3b7e4c73a6418311ad227aa2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146402/","number":"146402","commit":"c652ac83dc16fc62883120bc18a46e3f6195c442"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146439/","number":"146439","commit":"fd453a24cbaaaac390248960a656eeead0fe6eda"}]}
PIPELINE-->

qn895 committed:
[ML] Force click in APM deep link tests

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146455/)
* Commit: 2b02ba4eb79c32b938dba187e751cd264a27ae9c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-108860/2b02ba4eb79c32b938dba187e751cd264a27ae9c)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/19308576f333143bb4c7a43722d11e4ff66ff894) | [after](https://github.com/elastic/kibana/commit/2b02ba4eb79c32b938dba187e751cd264a27ae9c) | diff |
| --- | --- | --- | --- |
| `apm` | 4.4MB | 4.4MB | +2.4KB |

### History
* :broken_heart: [Build #146439](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146439/) failed fd453a24cbaaaac390248960a656eeead0fe6eda
* :broken_heart: [Build #146421](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146421/) failed 0536f2982733706c3b7e4c73a6418311ad227aa2
* :broken_heart: [Build #146402](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146402/) failed c652ac83dc16fc62883120bc18a46e3f6195c442
* :broken_heart: [Build #146379](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146379/) failed 3556cfb6adf3847aa3c6b028f031f94fc5ca673c
* :yellow_heart: [Build #146272](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146272/) was flaky d908b7ec7968e87f111a2b75b2955e9dda551929

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @walterra

<!--PIPELINE
{"builds":[{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146272/","number":"146272","commit":"d908b7ec7968e87f111a2b75b2955e9dda551929"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146379/","number":"146379","commit":"3556cfb6adf3847aa3c6b028f031f94fc5ca673c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146402/","number":"146402","commit":"c652ac83dc16fc62883120bc18a46e3f6195c442"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146421/","number":"146421","commit":"0536f2982733706c3b7e4c73a6418311ad227aa2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146439/","number":"146439","commit":"fd453a24cbaaaac390248960a656eeead0fe6eda"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146455/","number":"146455","commit":"2b02ba4eb79c32b938dba187e751cd264a27ae9c"}]}
PIPELINE-->

kibanamachine commented:
## 💚 Backport successful

| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [7.x](https://github.com/elastic/kibana/pull/109036)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/109036">](https://github.com/elastic/kibana/pull/109036) |

This backport PR will be merged automatically after passing CI.

walterra commented:
I added an item about the character encoding problem when filtering to the meta issue, it wasn't fixed as part of this PR, see #106381.
